Doc/library/ast.rst

```diff
@@ -1575,6 +1575,10 @@ and classes for traversing abstract syntax trees:
       sufficiently large/complex string due to stack depth limitations
       in Python's AST compiler.
 
+      It can raise :exc:`ValueError`, :exc:`TypeError`, :exc:`SyntaxError`,
+      :exc:`MemoryError` and :exc:`RecursionError` depending on the malformed
+      input.
+
```

Comment:
Just for my edification and for future reference, can you add a non-displayed in-line documentation comment showing an example of how each of these could happen.

Reply:
```py
import ast

recursion_tuple = ast.Tuple(elts=[], ctx=ast.Load())
recursion_tuple.elts.append(recursion_tuple)
INPUTS = [
    ("invalid syntax", SyntaxError),
    ("test", ValueError),
    (ast.List(elts=None, ctx=ast.Load()), TypeError),
    (recursion_tuple, RecursionError),
]

for inp, exc_type in INPUTS:
    try:
        ast.literal_eval(inp)
    except exc_type:
        continue
    else:
        assert False
```
does something like this suits?

Reply:
@rhettinger or maybe we can put this in an actual test in `test_ast` and refer it? How does that sounds?